### ISSUE TITLE
Is this project abandoned?

### ISSUE THREAD

**Issue by KristapsT** [JamesOsgood/mongodb-grafana#123]
Hey, pretty much what the title says.

This seems like a very useful alternative since the plugin for MongoDB datasource by Grafana is behind a paywall, but the last update to code seems to have been ~4 years ago and there are a bunch of open PRs going back 2-3 years.

It looks like the original author is no longer maintaining this, no new features or fixes to existing issues.

Anyone who is reliably using the plugin and is happy with it?

**Comment by BlackCodes** [comment]
I want to restart this project for every coder~

**Comment by cptechie** [comment]
I have started using this plugin today. I am relatively impressed that it still works, but much more improvements can be made.

I have published a [docker image](https://hub.docker.com/r/cptechie/mongodb_grafana) for this project, mostly so I can rapidly deploy this on many servers. Feel free to use it and check it out.

It is as simple as below
`docker run --name mongo_proxy --network host -d cptechie/mongodb_grafana`

Of if you rather not have --network host:
`docker run --name mongo_proxy --network host -d cptechie/mongodb_grafana`

Much less time consuming than installing it locally or waiting for a custom Dockerfile to build the image.

**Comment by KristapsT** [comment]
It works, but there are many issues that stop me from using this.

The main being that it leaks mongodb credentials to browser in plain text when you are inspecting queries in dashboard.

**Comment by ajeje93** [comment]
As @cptechie did I also published a docker container, but in my case it is the grafana docker image with the plugin preinstalled in the container: https://github.com/ajeje93/grafana-mongodb-docker

Another thing I noticed yesterday is that the proxy server does not work on node 17.
I did not noticed the leak that @KristapsT found mainly because I only used it for hobby projects and not for production environments.

**Comment by bygui86** [comment]
hi guys
which version of Grafana do you use?
I tried with Grafana v8.4.5 and it does not work :(

![Screenshot 2022-04-05 at 17 15 10](https://user-images.githubusercontent.com/22518997/161787093-95068718-995d-48bc-aed2-a2ac428168c6.png)

I used the test values given in the main readme:
- MongoDB URL: `mongodb://rpiread:rpiread@rpi-sensor-data-shard-00-00-ifxxs.mongodb.net:27017,rpi-sensor-data-shard-00-01-ifxxs.mongodb.net:27017,rpi-sensor-data-shard-00-02-ifxxs.mongodb.net:27017/test?ssl=true&replicaSet=rpi-sensor-data-shard-0&authSource=admin`
- MongoDB Database: `rpi`

I tried also with another MongoDB URL but nothing.

I deployed everything to k8s v.1.21.x, I installed mongodb-grafana plugin manually, here below info about it
![Screenshot 2022-04-05 at 17 17 08](https://user-images.githubusercontent.com/22518997/161787444-6df141d4-dc15-4851-a003-15799e083382.png)

please help!! :)

**Comment by bygui86** [comment]
I noticed there is an overlapping between MongoDB official enterprise plugin and this one... someone knows how can I solve it? Maybe that's the reason why it does not work...

**Comment by JamesOsgood** [comment]
Hello!

I developed this plugin shortly after joining MongoDB 4.5 years ago and TBH
I haven't done anything with it since then. The official one came
afterwards....

If anyone wants to take it over, I'd be more than happy.

James

**Comment by ajeje93** [comment]
> I noticed there is an overlapping between MongoDB official enterprise plugin and this one... someone knows how can I solve it? Maybe that's the reason why it does not work...

Yeah you should rename the plugin in file plugin.json (dist folder) to make it work because it has the same id of the official one.

**Comment by sanderp42** [comment]
Hi all, 

I am heavy user of the plugin (and my startup) and actually it still works good also with the latest grafana versions. Some functions don't work unfortunately such as alerting. 

Would be great if this project could be re-started. I am also willing to sponsor some development with some modest contributions. PM me if you are interested. 

@JamesOsgood, thanks a lot for building it! Would you consider renewing the plugin for sponsorship? Contact me on discord if you interested (Plyum#1720)

**Comment by bygui86** [comment]
@ajeje93 thanks for your suggestion, unfortunately I'm not able to make it work :(
I keep receiving the message `Bad gateway`
Any idea?

**Comment by soringabriel** [comment]
@bygui86 not sure if this helps or not, but I used to get the same error because I skipped [this part of installation](https://github.com/JamesOsgood/mongodb-grafana#install-and-start-the-mongodb-proxy-server )

After I started the mongodb proxy server everything worked correctly for me

**Comment by bygui86** [comment]
@soringabriel Thanks a lot for the suggestion!
I missed completely that part! I never run the proxy server :(
As soon as I can I will try again

**Comment by Breee** [comment]
We don't want to buy an enterprise plugin ;) 
If you need assistance to get everything working again, just let us know how.

**Comment by meln5674** [comment]
To those monitoring this thread: I investigating forking this and restarting development, but I found that there have been several improvements to Grafana's plugin API int he intervening time, notably there is now a feature that means there is no longer a requirement for the proxy server. As such, I have started an entirely new plugin using the new plugin API and the official mongodb driver library, which you can find here: https://github.com/meln5674/grafana-mongodb-community-plugin . I have only so far re-implemented timeseries queries, as well as made the credentials into secret fields so they are not leaked. I have a project that needs a plugin like this, so there will be semi-regular development for the near future.

**Comment by bygui86** [comment]
@meln5674 that's amazing! thanks a lot for the effort! I will try it soon :)

**Comment by emrys90** [comment]
I'm in the same boat of wanting to use MongoDB on a self hosted installation, and don't want to pay the expensive cost of the enterprise version.
